README.md

```diff
@@ -0,0 +1,211 @@
+      - [DELETE /session](#delete-session)
+  - [Security considerations](#security-considerations)
+  - [Branch naming convention](#branch-naming-convention)
+
+## Frontend
+
+Typical single page application.
+
+### Assets server
+
+> SAFETY NOTE:
+> The server should serve assets via [HTTPS](https://en.wikipedia.org/wiki/HTTPS).
+> I will not implement that.
+
+![spa](/images/spa.png)
+
+### Tooling
+
+- [react](reactjs.org)
+- [react-router](https://reactrouter.com/)
+- [typescript](https://www.typescriptlang.org/)
+- [styled-components](https://styled-components.com/)
+- [vite](https://vitejs.dev/)
+- [testing-library](https://testing-library.com/)
+- [nginx](https://www.nginx.com/)
+- [docker](https://www.docker.com/)
+
+### Routes
+
+#### /
+
+Redirects user to `/folder`.
+
+#### /login
+
+A login screen where an unauthenticated user is automatically redirected to (and then taken back to the original URL).
+
+#### /not-found
+
+Not found page, where a user is automatically redirected to in case folder, does not exist.
+
+#### /folder/\*\*/\*
+
+The page displays single folder content.
+
+### Design
+
+See [figma](https://www.figma.com/file/J6yvOILo6HM62FnHXaAAOC/FolderApp?node-id=0%3A1)
+
+- Based on google drive (material design).
+- Icons by [Icons8](https://icons8.com/)
+
+## Backend
+
+Folder data service. In-memory session authentication.
+
+### Tooling
+
+- [rust](https://www.rust-lang.org/)
+- [rocket](https://rocket.rs/)
+- [serde](https://serde.rs/)
+- [docker](https://www.docker.com/)
+
+### API
+
+> SAFETY NOTE:
+> In case we will use real DB to store data we should validate input against some kind of query injection (like SQL injection).
+
+#### GET /folder/\*\*/\*
+
+Returns folder data.
+
+- **responds** with:
+
+  - <span style="color:red;">error</span>, when a user is unauthenticated,
+
+    ```ts
+    HTTP/1.1 403 Forbidden
+    ```
+
+  - <span style="color:red;">error</span>, when a user is authenticated and the requested folder, doesn't exist,
+
+    ```ts
+    HTTP/1.1 404 Not found
+    ```
+
+  - <span style="color:green;">folder data</span> when the user is authenticated and the requested folder exists. A folder structure is always 1 level deep.
+
+    ```ts
+    HTTP/1.1 200 Ok
+
+    {
+        name: 'bar',
+        items: [
+            {
+                type: 'dir',
+                sizeKb: 0,
+                name: string
+            },
+            {
+                type: 'file',
+                sizeKb: number,
+                name: string
+            }
+        ]
+    }
+    ```
+
+#### POST /session
+
+Login.
+
+- **accepts**:
+
+  ```ts
+  {
+      name: string,
+      password: string
+  }
+  ```
+
+- **responds** with:
+
+  - <span style="color:red;">error</span> when credentials don't match any user,
+
+    ```ts
+    HTTP/1.1 400 Bad request
+    ```
+
+  - <span style="color:green;">success</span> code when the session is created. The token cookie is valid for one day.
```

Comment:
What will you use for a token?

Reply:
I would use JWTs stored in the cookies. JWT Public claims will contain the user's id.

Reply:
info added in 47d915443b7157670a40ff81c6a85387969469a1

Reply:
please also check f3048fdf8aba73e8be3c5f7229ae9a1d85007bea